### DISCUSSION TITLE
Image inpainting  - Image removal using a mask generated externally

### DISCUSSION THREAD

**Discussion by swarna-a-26** [Sanster/IOPaint#333] [General]
Hi,
First of all, thank you for the awesome Lama cleaner and sharing the code.
I wanted to remove an object from the image. I installed your code base locally and used your website tool to  remove the object using the image inpainting method. It worked great. I downloaded the generated mask for the image. I used the downloaded mask and made code changes to use the mask file as an external input and used the same image and checked. It worked fine and the object was removed. However, I generated another image mask externally using a different tool for the same image and provided that as an input  to the inpainting tool. But, that does not clean up the object properly.Attaching the picture, both working and not working image masks. I am not sure what the problem is with the new image mask. 

Working mask generated via lama cleaner - hippo_and_other_mask_working.jpg
Mask generated externally that is not working - mask_hippo_ipad_not_working.png
Test image- hippo_and_other.jpeg

Any thoughts why when I use mask_hippo_ipad_not_working.png mask, the object is not removed properly? 

  
![mask_hippo_ipad_not_working](https://github.com/Sanster/lama-cleaner/assets/2103865/02558a65-5539-4a0f-8e24-9b59a781e11f)
![hippo_and_other](https://github.com/Sanster/lama-cleaner/assets/2103865/c39b066e-8d37-4abc-9a72-96065ef7f077)

![hippo_and_other_mask_working](https://github.com/Sanster/lama-cleaner/assets/2103865/d9785cf2-be3b-4615-b205-9a8ef74309cd)

**Comment by Sanster** [comment]
It may be because your mask is too tight. For inpainting models, you need to dilate the mask slightly
